GitCommands/Submodules/SubmoduleStatusProvider.cs

```diff
@@ -46,76 +60,98 @@ public void Dispose()
         public void Init()
         {
             // Cancel any previous async activities:
-            var cancelToken = _submodulesStatusSequence.Next();
-            _previousSubmoduleHadChanges = false;
+            _submodulesStatusSequence.Next();
         }
 
-        public bool HasChangedToNone([CanBeNull] IReadOnlyList<GitItemStatus> allChangedFiles)
-        {
-            if (allChangedFiles == null)
-            {
-                return false;
-            }
-
-            bool anyUpdate = allChangedFiles.Any(i => i.IsSubmodule && (i.IsChanged || !i.IsTracked));
-
-            // If status is changed to none, the status must be cleared
-            bool result = _previousSubmoduleHadChanges && !anyUpdate;
-            _previousSubmoduleHadChanges = anyUpdate;
-            return result;
-        }
-
-        public bool HasStatusChanges([CanBeNull] IReadOnlyList<GitItemStatus> allChangedFiles)
-        {
-            TimeSpan elapsed = DateTime.Now - _previousSubmoduleUpdateTime;
-
-            // Throttle updates triggered from status updates
-            if (allChangedFiles == null || elapsed.TotalSeconds <= 15)
-            {
-                return false;
-            }
-
-            // If any submodules are changed, trigger an update
-            // TBD: This should check the status against last updated list and only update the required modules
-            // (maybe even ignore count)
-            return allChangedFiles.Any(i => i.IsSubmodule && (i.IsChanged || !i.IsTracked));
-        }
-
-        public void UpdateSubmodulesStatus(bool updateStatus, string workingDirectory, string noBranchText)
+        /// <inheritdoc />
+        public void UpdateSubmodulesStructure(GitModule gitModule, string noBranchText, bool updateStatus)
```

Comment:
⚠️ `GitModule` should not be passed as an input parameter for a number of reasons.
1. It is a combination of state and behaviours, behavioural components should not be passed around.
2. Passing a concrete implementation creates a strong coupling, this reduces cohesion and negatively affects testability
3. Passing an object with a massive public surface in order to get a single property is also a bad practice.

Please rework the API in a way that does not require passing neither `IGitModule` nor `GitModule`.

Reply:
This will require some calls to Git for data that already exists.
OK to use an optimized copy from #6373?

Reply:
API changed (without  #6373 for now)